SU-SWS/decanter
# 282: Fix modular-spacing mixins and variables

# READY

# Summary
- Fix modular-spacing related mixins, functions and variables so the values match the Figma file

# Needed By (Date)
- N/A

# Urgency
- Not urgent

# Steps to Test
- pull this branch and run grunt styleguide
- check that the variables in variables/core/_modular-spacing.scss are correct and the mixins/functions under modular-scale (the spacing related ones) work as intended (Figma documentation under Stanford Design Library -> Visual Guidelines -> Spacing)
-  _card.scss, _card--horizontal.scss now uses modular-padding mixin instead of old $modular-spacing variables - please check if that's ok
- core/scss/core/flex-grid/_flex-grid.scss now uses $grid-gutter values instead of old $modular-spacing variables - please check if that's ok

# Affected Projects or Products
- Decanter

# Associated Issues and/or People
- Affected components - cards, flex-grid (ones that use modular-spacing, modular-margins etc @sherakama @josephgknox )
- Note: Modular spacing and modular typography use totally different system to calculate the steps, even though they share the same base value at each breakpoint.
- Modular typography follows the Bourbon modular-scale mixin formula with a 1.25 scale ratio between each step, ie -
modular-scale(0) = $base-value
modular-scale(1) = $base-value * 1.25
modular-scale(2) = $base-value * 1.25 * 1.25 etc
- Modular spacing does NOT use the above system. There is no fix scale ratio value. Instead, the scale is manually adjusted by John and Kerri because they needed more granular control over the spacing values. This scale is documented in Figma.

Also,

- Modular spacing, screen margins, gutters are unrelated
